Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs

```diff
@@ -710,5 +735,61 @@ private void RetestSubmission(int submissionId)
             entry.Property(pr => pr.Processed).IsModified = true;
             entry.Property(pr => pr.Processing).IsModified = true;
         }
+
+        private DetailedProblemViewModel PrepareProblemViewModelForEdit(int id)
+        {
+            return this.Data.Problems.All()
+                .Where(x => x.Id == id)
+                .Select(DetailedProblemViewModel.FromProblem)
+                .FirstOrDefault();
+        }
+
+        private DetailedProblemViewModel PrepareProblemViewModelForCreate(Contest contest)
+        {
+            var lastOrderBy = -1;
+            var lastProblem = this.Data.Problems.All().Where(x => x.ContestId == contest.Id);
+
+            if (lastProblem.Any())
+            {
+                lastOrderBy = lastProblem.Max(x => x.OrderBy);
+            }
+
+            var problem = new DetailedProblemViewModel
+            {
+                Name = "Име",
+                MaximumPoints = 100,
+                TimeLimit = 100,
+                MemoryLimit = 16777216,
+                AvailableCheckers = this.Data.Checkers.All()
+                    .Select(checker => new SelectListItem
+                    {
+                        Text = checker.Name,
+                        Value = checker.Name,
+                        Selected = checker.Name.Contains("Trim")
+                    }),
+                OrderBy = lastOrderBy + 1,
+                ContestId = contest.Id,
+                ContestName = contest.Name,
+                ShowResults = true,
+                SourceCodeSizeLimit = 16384,
+                ShowDetailedFeedback = false,
```

Comment:
If this is not nullable, you can skip this line, because false is the default value of bool types.